javascripts/discourse/components/site-palette-menu-item.gjs

```diff
@@ -1,40 +1,52 @@
 import Component from "@glimmer/component";
-import { fn } from "@ember/helper";
 import { action } from "@ember/object";
 import { service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
 import DButton from "discourse/components/d-button";
 import concatClass from "discourse/helpers/concat-class";
+import { reload } from "discourse/helpers/page-reloader";
+import {
+  loadColorSchemeStylesheet,
+  updateColorSchemeCookie,
+} from "discourse/lib/color-scheme-picker";
 
 export default class SitePaletteMenuItem extends Component {
-  @service customColor;
+  @service site;
+  @service session;
+  @service interfaceColor;
 
   get siteStyle() {
-    return `--icon-color: ${this.args.colorPalette.color}`;
-  }
-
-  get activeClass() {
-    if (this.customColor.color === this.args.colorPalette.name) {
-      return "active";
-    }
+    return `--icon-color: ${this.args.colorPalette.accent}`;
   }
 
   @action
-  handleInput(colorPalette) {
-    this.customColor.setColor(colorPalette.name);
+  handleInput() {
+    console.log(this.args.colorPalette);
+
+    if (this.interfaceColor.lightModeForced) {
+      loadColorSchemeStylesheet(this.args.colorPalette.id);
+      updateColorSchemeCookie(this.args.colorPalette.id);
+      updateColorSchemeCookie(this.args.colorPalette.correspondingDarkModeId, {
+        dark: true,
+      });
+    } else if (this.interfaceColor.darkModeForced) {
+      loadColorSchemeStylesheet(this.args.colorPalette.correspondingDarkModeId);
+      updateColorSchemeCookie(this.args.colorPalette.id);
+      updateColorSchemeCookie(this.args.colorPalette.correspondingDarkModeId, {
+        dark: true,
+      });
+    }
+    reload();
   }
 
   <template>
-    <div class="color-palette-menu__item" data-color={{@colorPalette.name}}>
+    <div class="color-palette-menu__item" data-color="color-palette-name">
```

Comment:
This change doesn't make sense, we should be keeping the actual color name in the attr here, otherwise the attr is the same for all of them